WordPress/gutenberg
# fix: Enable access to block settings within UBE

## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/5508



## What?

Repair access to the block settings sidebar within the unsupported block editor
(UBE). Fixes #48298. 

Repair hiding of unrelated editor UI in the UBE. Fixes #47966. 

## Why?

Without access to the block settings, a user cannot modify important settings
for a given block. 

Also, displaying UI elements unrelated to the sole block is
confusing and unnecessary. 

## How?


### Block Settings
Previously, the UBE relied upon the presence of the "Show more settings" button,
but that was removed in #46709. This updates the UBE to rely upon the sidebar
toggle button that is present in the post editor header. Because the sidebar toggle 
button can also reveal the post settings, this only displays the sidebar toggle 
button whenever a block is selected.

### Post Title Visibility

The override styles used by the UBE were no longer successfully applied once https://github.com/WordPress/gutenberg/pull/46212 
wrapped the editor within an `iframe`. It is not possible to style `iframe` 
children elements from the parent `document` context. These changes no 
copy the override styles to reside within the `iframe` document so they are 
applied to the children elements. 

### Unrelated Changes

This also updates existing Android-specific scripts to only run when Android
devices are detected by parsing the user agent string. Previously, the entire
`editor-behavior-overrides` script was only loaded by the Android platform. 
Now, both platforms load the script, and portions of it only run for Android. 

## Testing Instructions





> **Note**
> There is one known issue. On iOS, tapping towards the bottom of the screen — between the sole block and the keyboard toolbar — clears the block selection, but the sidebar toggle button remains erroneously displayed. This does not occur for Android. This can be seen at the end of the screencast below. 

Verify the steps outlined in #48298, #34668, and #47966 pass.

1. Using the WordPress mobile app, open a post containing unsupported blocks,
   e.g. Archives, Table.
1. Select an unsupported block.
1. Tap the <kbd>?</kbd> button.
1. Tap "Edit using web editor."
1. Verify the sidebar toggle button is not present in the top-right corner.
1. Select the unsupported block.
1. Tap the sidebar toggle button in the top-right corner.
1. Verify the block settings are displayed.

### Testing Instructions for Keyboard

n/a

## Screenshots or screencast 

https://user-images.githubusercontent.com/438664/221731846-229b3e08-321d-417f-aa4e-68b1338ec86e.mov
